articles/list-transformations-js/index.md

```diff
@@ -0,0 +1,357 @@
+//{ name: 'Toyota', origin: 'japan
+```
+In the code above:
+1. We declare a function `allCars()` with a parameter `brands`.
+2. Inside it we create a [`for()` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for) that loops through `brands` and logs them.
+3. We call the function `allCars()`and pass an argument `cars.brands`.
+
+### Looping through nested objects using a `for()` loop
+
+We can get common properties in nested objects using a for loop. Let us log all the `name` properties of the nested brand objects.
+
+```javascript
+function names(brands){
+    for (var i = 0; i < brands.length; i++){
+        console.log(brands[i].name)
+    };
+};
+
+names(cars.brands); 
+//Ford
+//Toyota
+```
+In the above code:
+1. We declare a function `names()` with a parameter `brands`.
+2. We create a for loop that logs each `name` property of the nested objects.
+3. We then call the `names()` function and pass an argument `cars.brands`.
+
+### Looping through nested objects properties using a `for ...in` loop
+
+If we want to get all the properties of a nested object, we can use a [`for ...in`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) loop. Let's loop through all the properties of the nested `brands` array.
+
+```javascript
+function everything (brands){
+    for(var i =0; i< brands.length; i++){
+        for (var property in brands[i]){
+            console.log(brands[i][property])
+        }
+    }
+};
+
+everything(cars.brands); 
+//Ford
+//USA
+//Toyota
+//Japan
+```
+In the above code:
+1. We declare a function `everything()` with a parameter `brands`.
+2. Inside it, we create a [`for()` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for) that loops through all the items in the `brands` array.
+3. Inside it, we create a [`for ...in`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) loop that loops through all the object properties.
+The for-in loop, assigns each [enumerable property](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties) of the object to the variable `key` and then logs it.
+4. We call the function `everything()` and pass an argument `cars.brands`.
+
+### Destructuring a nested data structure
+
+Destructuring can be used to extract data from nested objects and arrays. We can extract the two `brands` objects we created earlier into two new variables.
+```javascript
+const {
+  brands: [
+      brand1,
+	  brand2
+	]
+} = cars;
+
+console.log(brand1, brand2); //{name: 'Ford', origin: 'USA'} {name: 'Toyota', origin: 'Japan'}
+```
+We can also extract the `name` and `origin` properties of the **first** `brands` array.
+```javascript
+const [
+    {
+      name,
+      origin
+   }
+  ] = cars.brands;
+
+console.log(name, origin) //Ford USA
+```
+We can also extract all the `name` and `origin` properties of the `brands` array.
```

Comment:
add
"... properties of the objects in the `brands` array."
Similar to line149 I think it was an ommision.